js/initialize.jsx

```diff
@@ -0,0 +1,147 @@
+'use strict'
+
+var React = require('react/addons')
+var ipc = window.require('remote').require('ipc')
+
+var $ = require('jquery-bf')
+var bs = require('bootstrap')
+
+var Initform = React.createClass({
+  displayName: 'Initform',
+  mixins: [React.addons.LinkedStateMixin],
+
+  getInitialState: function () {
+    return {
+      error: null,
+      directory: '',
+      initState: 'uninitialized',
+      keysize: 2048
+    }
+  },
+
+  componentDidMount: function () {
+    var t = this
+    ipc.on('default-directory', function (dir) {
+      t.setState({directory: dir})
+    })
+
+    ipc.on('initializing', function () {
+      t.setState({initState: 'initializing'})
+    })
+
+    ipc.on('initialization-error', function (err) {
+      t.setState({
+        error: err,
+        initState: 'uninitialized'
+      })
+    })
+
+    ipc.on('initialization-complete', function () {
+      t.setState({initState: 'initialized'})
+    })
+  },
+
+  setKeySize: function (size) {
+    var t = this;
```

Comment:
Semicolon, also if we are writing this with standard, `this` should be assigned to `var self` correct?